src/NServiceBus.Core/SettingsExtensions.cs

```diff
@@ -30,16 +31,30 @@ public static string EndpointName(this IReadOnlySettings settings)
         /// <summary>
         /// Returns the transport specific address of the shared queue name of this endpoint.
         /// </summary>
+        [ObsoleteEx(
+            Message = "Inject the ReceiveAddresses class to access the endpoint's receiving transport addresses at runtime. See the NServiceBus version 8 upgrade guide for further details.",
+            TreatAsErrorFromVersion = "9",
+            RemoveInVersion = "10")]
         public static string LocalAddress(this IReadOnlySettings settings)
         {
             Guard.AgainstNull(nameof(settings), settings);
 
+            if (!settings.TryGet<TransportDefinition>(out var transportDefinition))
```

Comment:
do we need this check at all? It isn't possible to configure an endpoint without transport IIRC and you can't access LocalAddress before the endpoint is configured/started which is done by the next check?